src/core/gui/XournalView.cpp

```diff
@@ -717,10 +722,13 @@ void XournalView::documentChanged(DocumentChangeType type) {
     for (auto&& page: viewPages) { delete page; }
     viewPages.clear();
 
-    this->cache->clearCache();
+    this->cache.reset();
 
     Document* doc = control->getDocument();
     doc->lock();
+    if (doc->getPdfPageCount() != 0) {
+        cache = std::make_unique<PdfCache>(doc->getPdfDocument(), control->getSettings()->getPdfPageCacheSize());
```

Comment:
Do we also need to update the PdfCache's size when the settings change?

Reply:
It'd be better indeed. As it is, the cache size is only applied to newly opened documents.
This is a papercut, but there is another issue with the settings: see other comment (above or below, I don't know :)